### PR TITLE
Add local pytorch model and cli-produced-model support for cli

### PR DESCRIPTION
**Pull request by xiaoyu-work**
## Describe your changes

Add local pytorch model and cli-produced-model support for cli.

## Checklist before requesting a review
- [x] Add unit tests for this change.
- [x] Make sure all tests can pass.
- [x] Update documents if necessary.
- [x] Lint and apply fixes to your code by running `lintrunner -a`
- [ ] Is this a user-facing change? If yes, give a description of this change to be included in the release notes.
- [ ] Is this PR including examples changes? If yes, please remember to update [example documentation](https://github.com/microsoft/Olive/blob/main/docs/source/examples.md) in a follow-up PR.

### CODE REVIEW

File: olive/cli/base.py [hunk 1]
```diff
@@ -39,25 +40,145 @@ def run(self):
...
+    }
+
+    if args.script_dir:
+        input_model_config["script_dir"] = args.script_dir
+
+    # model path has high priority than model loader
+    if model_path:
+        input_model_config["model_path"] = model_path
+    else:
+        if not user_module_loader.has_function("_model_loader"):
```
Comment (jambayk) [inline olive/cli/base.py:84]: There can be cases where both model_path and model_loader are present. For example, a user has a custom model architecture and a weights file. they would need both.

Comment (xiaoyu-work) [reply olive/cli/base.py]: Good call! Added this case support.

File: olive/model/handler/mixin/dummy_inputs.py [hunk 2]
```diff
@@ -15,16 +15,11 @@ class DummyInputsMixin:
     the dummy data is used to evaluate the latency if user doesn't provide the data for evaluation.
     """
 
-    def _get_dummy_inputs_from_io_config(self, force_kv_cache: bool = False, filter_hook=None, filter_hook_kwargs=None):
+    def _get_dummy_inputs_from_io_config(self, filter_hook=None, filter_hook_kwargs=None):
```
Comment (jambayk) [inline olive/model/handler/mixin/dummy_inputs.py:18]: please undo this change. this argument is used by the hf model handler https://github.com/microsoft/Olive/blob/0786cd091a745cc7eb5f1ede22527d2f235fff99/olive/model/handler/hf.py#L116

File: olive/model/handler/pytorch.py [hunk 3]
```diff
@@ -196,9 +196,7 @@ def get_dummy_inputs(self, filter_hook=None, filter_hook_kwargs=None):
             # respect user's dummy_inputs_func, no hook
             return user_module_loader.call_object(self.dummy_inputs_func, self)
 
-        dummy_inputs = self._get_dummy_inputs_from_io_config(
-            filter_hook=filter_hook, filter_hook_kwargs=filter_hook_kwargs
-        )
+        dummy_inputs = self._get_dummy_inputs_from_io_config(filter_hook_kwargs=filter_hook_kwargs)
```
Comment (jambayk) [inline olive/model/handler/pytorch.py:199]: why is filter_hook argument removed here?

File: olive/model/handler/mixin/dummy_inputs.py (hunk 2 above)
Comment (jambayk) [reply olive/model/handler/mixin/dummy_inputs.py:18]: synced offline

**Review by jambayk — APPROVED** [review]

**Review by jambayk — APPROVED** [review]
